### ISSUE TITLE
Forbid the usage of property

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#5122]
## Expected Behavior of the rule
Forbid the usage of a property like the same way we have `ForbiddenMethodCall`.

## Context
When you are using functions like `combine`, `zip` or similar ones `Pair` and `Triple` are really handy classes if then you continue using it with destructuring declarations. BUT if you access it data using `first`, `second` or `third` those classes are really bad because they give you 0 information about what you are accessing.

For that reason I think that forbid the usage of `Pair.first`, `Pair.second`, `Triple.first`... would be beneficial.

Example:

This is ok:

```kotlin
listOf("Robb", "Sansa", "Arya", "Brandon", "Rickon", "John")
  .zip("Grey Wind", "Lady", "Nymeria", "Summer", "Shaggy Dog", "Ghost")
  .map { (child, wolve) -> println("$child owns $wolve") }
```

This is not ok

```kotlin
listOf("Robb", "Sansa", "Arya", "Brandon", "Rickon", "John")
  .zip("Grey Wind", "Lady", "Nymeria", "Summer", "Shaggy Dog", "Ghost")
  .map { println("${it.first} owns ${it.second}") }
```

The reason of this proposal: https://github.com/detekt/detekt/pull/5058#discussion_r927946635

Related issue: #5015 we talked a bit about this topic there.

**Comment by cortinico** [comment]
Closing as this was discussed here:
https://github.com/detekt/detekt/issues/5015
and effectively implement
https://github.com/detekt/detekt/pull/5078
